src/stash-list/bookmark.vue

```diff
@@ -102,6 +103,12 @@ export default defineComponent({
             await this.model().deleteBookmark(this.bookmark);
         })},
 
+        closeTabs(ev: MouseEvent) { this.model().attempt(async () => {
+            await this.model().tabs.remove(this.related_tabs
+                .filter((t) => ! t.hidden && t.windowId === this.targetWindow)
```

Comment:
Use `model().hideOrCloseStashedTabs()` instead; this will ensure tabs stay hidden (according to the user's preferences) so long as they're still in the stash.

Reply:
> Use `model().hideOrCloseStashedTabs()` instead

I can do this, but I'm not sure it really makes sense to me that it would behave differently from something like just hitting the hotkey to close a tab. It also doesn't behave like stashing an open tab that already had a stash bookmark (which I haven't tried until now). I'd expect this to just close the tab, but it actually creates a duplicate bookmark.

Reply:
>I can do this, but I'm not sure it really makes sense to me that it would behave differently from something like just hitting the hotkey to close a tab

As far as the user is concerned, it doesn't, except that stashed tabs will (by default) be restored more quickly.

Let me add a little more context (some of which you may already know, but bear with me):

1. `bookmark.vue` is only ever used to represent a stashed tab (Tab Stash doesn't care about things outside of the stash).
2. Tabs that get stashed are either closed or hidden, according to the user's preference.
3. `hideOrCloseStashedTabs()` implements the "close/hide" part of "stash and then close or hide".

In general, the invariant is, "if it's in the stash and there's a tab for it somewhere, we should hang onto that tab per user preferences". By default, this means hiding the tab rather than closing it--that way it can more quickly be restored later, we save what the user is working on, etc. (Of course, this isn't a strict rule--there are plenty of ways hidden tabs can get closed, e.g. the user explicitly closing the tab in the browser tab bar, closing the whole window, etc.)

This applies to closing tabs, too (at least insofar as Tab Stash has control over this). For example, there is the "Close all stashed tabs" button (in "Unstashed Tabs"):

https://github.com/josh-berry/tab-stash/blob/e0b0214f1da1526b9ef4920c73b979ce895317d5/src/stash-list/window.vue#L174

Apart from a bit of cruft I noticed just now (you can ignore the `if` statement on line 175, it's harmless but I just now noticed it's redundant), this does almost exactly what you're trying to do here.

> It also doesn't behave like stashing an open tab that already had a stash bookmark (which I haven't tried until now). I'd expect this to just close the tab, but it actually creates a duplicate bookmark.

Yeah, de-duplicating is done with a light touch--we (mostly) try to avoid having duplicates in the same folder but otherwise leave them alone. I don't want to be *too* prescriptive about de-duplication because I can imagine someone *wanting* duplicates in some specific scenario, and having them isn't more than a minor inconvenience.

Reply:
I appreciate the detailed reply! I personally would never want to just hide open tabs in that scenario but I can just set the stash behavior to close immediately.

As for duplicates, it could be interesting to add a setting for that at some point. This would also allow stashing to be used to just generally indicate being done with a tab that you may want to return to later.

One thing I've also been looking at is context menu behavior. Currently, you just get the context menu for a link which is somewhat limited. Overriding the context menu could provide more relevant functions for tabs/bookmarks to allow stuff like duplicating them without adding button overload.

Anyway, I've made those two changes and haven't noticed any regressions.